### PR TITLE
Add optional inner system view connector coordinates

### PR DESCRIPTION
**Pull request by peter-lobner**
A pragmatic minor change as discussed with @klausschuch to support optional connectors for the system inner view to enable to preserve the inside system layout when integrating a system in a system structure and to avoid unintended changes to the position of the connectors on the inside view of the system when doing layout changes on the outside view and vice versa.

### CODE REVIEW

**Comment by JochenKoehler** [comment]
Web meeting 2024-04-26
- Some remarks from Pierre regarding "cosmetic changes". Will do the changes accordingly
- Dag is asking if his topic regarding rotating of connectors was taken into account. This is not the case. Dag will handle this eventually with a separate pull request (maybe for a next release)

File: docs/5___ssd.adoc [hunk 1]
```diff
@@ -166,13 +166,19 @@ Note that x and y coordinates are in a special coordinate system, where 0,0 is t
 
 For systems the placement of connectors for the inside and outside view of the system is identical, the special coordinate system is just translated to different actual coordinate systems, namely the one determined by the ElementGeometry for the outside view, and the one determined by SystemGeometry for the inside view.
 
+For systems by default the placement of connectors for the inside and outside view of the system is identical, the special coordinate system is just translated to different actual coordinate systems, namely the one determined by the ElementGeometry for the outside view, and the one determined by SystemGeometry for the inside view.
+
+For systems optionally a different placement of connectors for the inside view of the system can be specified with the systemInnerX and systemInnerY coordinates. This enables to preserve the inside system layout when integrating a system in a system structure and to avoid unintended changes to the position of the connectors on the inside view of the system when doing layout changes on the outside view and vice versa.
```
Comment (pmai) [inline docs/5___ssd.adoc:171]:
```suggestion
By default, the placement of connectors for a system's inside and outside views is identical.
The special coordinate system is just translated to different actual coordinate systems, namely the one determined by ElementGeometry for the outside view and the one determined by SystemGeometry for the inside view.

For systems, optionally, different placement of connectors for the inside view of the system can be specified with the systemInnerX and systemInnerY coordinates.
This enables preserving the inside system layout when integrating a system in a system structure and avoiding unintended changes to the position of the connectors on the inside view of the system when making layout changes on the outside view and vice versa.
```

Removes duplication, splits up and slightly rewrites for clarity and grammar.

File: schema/SystemStructureDescription.xsd [hunk 2]
```diff
@@ -659,16 +659,26 @@
...
+                                    one determined by the ElementGeometry for the outside view, and the
                                     one determined by SystemGeometry for the inside view.
+                                    
+                                    For systems optionally a different placement of connectors for the
+                                    inside view of the system can be specified with the systemInnerX and
+                                    systemInnerY coordinates. This enables to preserve the inside system
+                                    layout when integrating a system in a system structure and to avoid
+                                    unintended changes to the position of the connectors on the inside view
+                                    of the system when doing layout changes on the outside view and vice
+                                    versa.
```
Comment (pmai) [inline schema/SystemStructureDescription.xsd:674]:
```suggestion
                                    By default, the placement of connectors for a system's inside and
                                    outside views is identical. The special coordinate system is just
                                    translated to different actual coordinate systems, namely the one
                                    determined by ElementGeometry for the outside view and the one
                                    determined by SystemGeometry for the inside view.
                                    
                                    For systems, optionally, different placement of connectors for the
                                    inside view of the system can be specified with the systemInnerX
                                    and systemInnerY coordinates. This enables preserving the inside
                                    system layout when integrating a system in a system structure and
                                    avoiding unintended changes to the position of the connectors on
                                    the inside view of the system when making layout changes on the
                                    outside view and vice versa.
```

File: schema/SystemStructureDescription11.xsd [hunk 3]
```diff
@@ -656,16 +656,26 @@
...
+                                    one determined by the ElementGeometry for the outside view, and the
                                     one determined by SystemGeometry for the inside view.
+                                    
+                                    For systems optionally a different placement of connectors for the
+                                    inside view of the system can be specified with the systemInnerX and
+                                    systemInnerY coordinates. This enables to preserve the inside system
+                                    layout when integrating a system in a system structure and to avoid
+                                    unintended changes to the position of the connectors on the inside view
+                                    of the system when doing layout changes on the outside view and vice
+                                    versa.
```
Comment (pmai) [inline schema/SystemStructureDescription11.xsd:671]:
```suggestion
                                    By default, the placement of connectors for a system's inside and
                                    outside views is identical. The special coordinate system is just
                                    translated to different actual coordinate systems, namely the one
                                    determined by ElementGeometry for the outside view and the one
                                    determined by SystemGeometry for the inside view.
                                    
                                    For systems, optionally, different placement of connectors for the
                                    inside view of the system can be specified with the systemInnerX
                                    and systemInnerY coordinates. This enables preserving the inside
                                    system layout when integrating a system in a system structure and
                                    avoiding unintended changes to the position of the connectors on
                                    the inside view of the system when making layout changes on the
                                    outside view and vice versa.
```

**Review by pmai — COMMENTED** [review]
Slight rewriting for clarity and grammar.

**Review by DagBruck — APPROVED** [review]
If inner and outer coordinates are different, a "transparent" view shall use the outer view's coordinates.

**Comment by JochenKoehler** [comment]
web meeting 2024-05-08: Klaus will have a final look at it and add some review comments for final merge

File: docs/5___ssd.adoc [hunk 4]
```diff
@@ -164,7 +164,11 @@ image:images/image28.png[image,width=307,height=149]
...
+By default, the placement of connectors for a system's inside and outside views is identical.
+The special coordinate system is just translated to different actual coordinate systems, namely the one determined by ElementGeometry for the outside view and the one determined by SystemGeometry for the inside view.
+
+For systems, optionally, different placement of connectors for the inside view of the system can be specified with the systemInnerX and systemInnerY coordinates.
+This enables preserving the inside system layout when integrating a system in a system structure and avoiding unintended changes to the position of the connectors on the inside view of the system when making layout changes on the outside view and vice versa.
 
 If defined, this ConnectorGeometry overrides any ConnectorGeometry of a System in a referenced SSD file or any port location defined by an FMU (as defined in the relevant FMI standards).
```
Comment (pmai) [inline docs/5___ssd.adoc:173]: F2F 2024-06-13: The intent should be as follows, wording and specific rules to be adjusted later:

```suggestion
If defined, this ConnectorGeometry overrides any ConnectorGeometry of a System in a referenced SSD file or any port location defined by an FMU (as defined in the relevant FMI standards) for the x and y coordinates of the outside view.
Any inside view is not overridden.
```

File: docs/5___ssd.adoc [hunk 5]
```diff
@@ -164,15 +164,22 @@ image:images/image28.png[image,width=307,height=149]
 This optional XML element gives the geometry information of the connector.
 Note that x and y coordinates are in a special coordinate system, where 0,0 is the lower-left corner of the containing model element, and 1,1 is the upper-right corner of the model element, regardless of aspect ratio.
```
Comment (pmai) [inline docs/5___ssd.adoc:165]:
```suggestion
Note that x and y coordinates are in a normalized connector coordinate system, where 0,0 is the lower-left corner of the containing model element, and 1,1 is the upper-right corner of the model element, regardless of aspect ratio.
```

**Review by pmai — APPROVED** [review]
CCB 2024-09-13: Can be merged once all the occurrences of special coordinate system are changed to connector coordinate system.

**Review by klausschuch — APPROVED** [review]

**Pull request merged**